tests/lib/rules/no-empty-function.js

```diff
@@ -386,3 +390,317 @@ ruleTester.run(
+			messageId: "unexpected",
+			data: { name: "setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "class A {static set foo(value: string) {}}",
+			messageId: "unexpected",
+			data: { name: "static setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "const A = class {set foo(value: string) {}};",
+			messageId: "unexpected",
+			data: { name: "setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "const A = class {static set foo(value: string) {}};",
+			messageId: "unexpected",
+			data: { name: "static setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "class A { constructor(param: string) {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "constructors",
+		},
+		{
+			code: "const A = class {constructor(param: string) {}};",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "constructors",
+		},
+		{
+			code: "const foo = { async method(param: string) {} }",
+			allow: "asyncMethods",
+			messageId: "unexpected",
+			data: { name: "async method 'method'" },
+		},
+		{
+			code: "async function a(param: string){}",
+			allow: "asyncFunctions",
+			messageId: "unexpected",
+			data: { name: "async function 'a'" },
+		},
+		{
+			code: "const foo = async function(param: string) {}",
+			messageId: "unexpected",
+			data: { name: "async function" },
+			allow: "asyncFunctions",
+		},
+		{
+			code: "class A { async foo(param: string) {} }",
+			messageId: "unexpected",
+			data: { name: "async method 'foo'" },
+			allow: "asyncMethods",
+		},
+		{
+			code: "const foo = async (): Promise<void> => {};",
+			messageId: "unexpected",
+			data: { name: "async arrow function" },
+			allow: "arrowFunctions",
+		},
+		{
+			code: "class A { private constructor() {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "privateConstructors",
+		},
+		{
+			code: "const A = class { private constructor() {} };",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "privateConstructors",
+		},
+		{
+			code: "class A { protected constructor() {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "protectedConstructors",
+		},
+		{
+			code: "const A = class { protected constructor() {} };",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "protectedConstructors",
+		},
+		{
+			code: "class A { @decorator() foo() {} }",
+			messageId: "unexpected",
+			data: { name: "method 'foo'" },
+			allow: "decoratedFunctions",
+		},
+		{
+			code: "const A = class { @decorator() foo() {} }",
+			messageId: "unexpected",
+			data: { name: "method 'foo'" },
+			allow: "decoratedFunctions",
+		},
+		{
+			code: "class A extends B { override foo() {} }",
+			messageId: "unexpected",
+			data: { name: "method 'foo'" },
+			allow: "overrideMethods",
+		},
+	].reduce(toValidInvalid, {
+		valid: [
+			{
+				code: "class A { constructor(private param: string) {} }",
```

Comment:
[Testing] IMO there should also be tests for other kinds of parameter properties:

* `public`
* `protected`
* `readonly`

They all happen to pass right now, but just to be safe...